modules/createBowerPackage.js

```
@@ -0,0 +1,55 @@
+import { join as joinPaths} from 'path'
+import { stat as statFile, readFile, createWriteStream } from 'fs'
+import getProperty from './getProperty'
+import archiver from 'archiver'
+
+
+const promisify = func => (...args) =>
```

I've been an advocate of promises for several years now. I've spoken about them at conferences, encouraged others to use them, and been an active participant in discussions around evolving the promise API. But I'm over it now, for a few reasons:

- If you're not extremely careful, it's easy to swallow errors. After years of using them, I still occasionally accidentally wrote code that swallowed errors.
- Promises force everything to happen asynchronously, which removes the option for code to run synchronously. Most of the time I use them, I'm not actually interested in the async guarantee that promises make. Instead, I'm more interested in just having a generic way to specify some code that needs to run later. Callbacks let me run my code either async or sync, which I like.